cli/npm/resolvers/common.rs

```diff
@@ -0,0 +1,89 @@
+use std::path::Path;
+use std::path::PathBuf;
+
+use deno_ast::ModuleSpecifier;
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use deno_core::futures;
+use deno_core::futures::future::BoxFuture;
+use deno_core::url::Url;
+
+use crate::npm::NpmCache;
+use crate::npm::NpmPackageId;
+use crate::npm::NpmPackageReq;
+use crate::npm::NpmResolutionPackage;
+
+/// Information about the local npm package.
+pub struct LocalNpmPackageInfo {
+  /// Unique identifier.
+  pub id: NpmPackageId,
+  /// Local folder path of the npm package.
+  pub folder_path: PathBuf,
+}
+
+pub trait InnerNpmPackageResolver: Send + Sync {
```

Comment:
Why "Inner" ? Usually I wouldn't expect that name exposed from a module

Reply:
I'm not sure about a name. It's exposed from this module, but private to `cli/npm/resolvers` (not exported from there).